src/pages/account/AccountPage.tsx

```diff
@@ -16,33 +17,73 @@ import {
 } from "../../ui-utils/states";
 import { Transfer, TransfersTable } from "./TransfersTable";
 
+type ICoin = codec.cosmos.base.v1beta1.ICoin;
+const { Tx } = codec.cosmos.tx.v1beta1;
+
 export function AccountPage(): JSX.Element {
-  const clientContext = React.useContext(ClientContext);
-  const { address: addressParam } = useParams();
+  const { launchpadClient, stargateClient, registry } = React.useContext(ClientContext);
+  const { address: addressParam } = useParams<{ readonly address: string }>();
   const address = addressParam || "";
 
-  const [account, setAccount] = React.useState<Account | undefined | ErrorState | LoadingState>(loadingState);
+  const [balance, setBalance] = React.useState<readonly ICoin[] | ErrorState | LoadingState>(loadingState);
   const [transfers, setTransfers] = React.useState<readonly Transfer[] | ErrorState | LoadingState>(
     loadingState,
   );
 
-  React.useEffect(() => {
-    clientContext.client
+  const launchpadEffect = (): void => {
+    launchpadClient
       .getAccount(address)
-      .then(setAccount)
-      .catch(() => setAccount(errorState));
-    clientContext.client
+      .then((account) => setBalance(account?.balance ?? []))
+      .catch(() => setBalance(errorState));
+    launchpadClient
       .searchTx({ sentFromOrTo: address })
-      .then((execTxs) => {
+      .then((txs) => {
         const out = new Array<Transfer>();
-        for (const tx of execTxs) {
+        for (const tx of txs) {
           for (const [index, msg] of tx.tx.value.msg.entries()) {
             if (isMsgSend(msg)) {
               out.push({
                 key: `${tx.hash}_${index}`,
                 height: tx.height,
                 transactionId: tx.hash,
-                msg: msg,
+                msg: {
+                  fromAddress: msg.value.from_address,
+                  toAddress: msg.value.to_address,
+                  amount: [...msg.value.amount],
+                },
+              });
+            } else {
+              // skip
```

Comment:
Maybe a `.filter` would be more eadable than this skip code?

Reply:
This pattern was copied from other parts of the code. I can update.

Reply:
Both ways are fine for me. Whatever you prefer is good.